plugins/amoeba/platforms/common/src/kernels/amoebaGk.cc

```diff
@@ -87,102 +87,87 @@ DEVICE real interpolate2D(real x1, real x2, real y1, real y2, real x, real y,
     real calculateIndex = (rho - MINIMUM_RADIUS) / SPACING;
-    int below = (int) floor(calculateIndex);
-    int above = below + 1;
+    *below = (int) floor(calculateIndex);
+    *above = *below + 1;
     int NUM_POINTS = 45;
 
-    if (above >= NUM_POINTS) {
+    if (*above >= NUM_POINTS) {
         // Extrapolate up from the top table values.
-        below = NUM_POINTS - 1;
-        above = NUM_POINTS - 2;
-    } else if (below < 0) {
+        *below = NUM_POINTS - 1;
+        *above = NUM_POINTS - 2;
+    } else if (*below < 0) {
         // If below is less than 0, extrapolate down from the bottom table values.
-        below = 0;
-        above = 1;
+        *below = 0;
+        *above = 1;
     }
-
-    bounds[0] = below;
-    bounds[1] = above;
 }
 
-DEVICE void getNeckConstants(real rhoDescreened, real rhoDescreening, real constants[],
+DEVICE void getNeckConstants(real rhoDescreened, real rhoDescreening, real* aij, real *bij,
     GLOBAL const float* RESTRICT neckRadii,
     GLOBAL const float* RESTRICT neckA,
     GLOBAL const float* RESTRICT neckB) {
 
     // Convert the input radii from nm to A.
-    rhoDescreened *= 10.0;
-    rhoDescreening *= 10.0;
+    rhoDescreened *= 10;
+    rhoDescreening *= 10;
 
     // Determine low and high values for integration
-    int boundsI[] = {0, 1};
-    getBounds(rhoDescreened, boundsI);
-
-    int boundsJ[] = {0, 1};
-    getBounds(rhoDescreening, boundsJ);
-
-    int lowI = boundsI[0];
-    int highI = boundsI[1];
-    int lowJ = boundsJ[0];
-    int highJ = boundsJ[1];
-
+    int lowI, highI, lowJ, highJ;
+    getBounds(rhoDescreened, &lowI, &highI);
+    getBounds(rhoDescreening, &lowJ, &highJ);
     // Interpolate/Extrapolate Aij and Bij constant values
     int NUM_POINTS = 45;
     int lowOff = lowI * NUM_POINTS;
     int highOff = highI * NUM_POINTS;
-    real aij = interpolate2D(neckRadii[lowI], neckRadii[highI], neckRadii[lowJ], neckRadii[highJ],
+    *aij = interpolate2D(neckRadii[lowI], neckRadii[highI], neckRadii[lowJ], neckRadii[highJ],
                                rhoDescreened, rhoDescreening,
                                neckA[lowOff + lowJ], neckA[highOff + lowJ],
                                neckA[lowOff + highJ], neckA[highOff + highJ]);
 
-    real bij = interpolate2D(neckRadii[lowI], neckRadii[highI],neckRadii[lowJ], neckRadii[highJ],
+    *bij = interpolate2D(neckRadii[lowI], neckRadii[highI],neckRadii[lowJ], neckRadii[highJ],
                                rhoDescreened, rhoDescreening,
                                neckB[lowOff + lowJ], neckB[highOff + lowJ],
                                neckB[lowOff + highJ], neckB[highOff + highJ]);
 
     // Never let Aij be negative.
-    if (aij < 0.0) {
-        aij = 0.0;
+    if (*aij < 0.0f) {
+        *aij = 0.0f;
     }
 
     // Convert aij from A^(-11) to nm^(-11);
-    constants[0] = aij * 1.0e11;
+    *aij *= 1.0e11f;
     // Convert bij from A to nm.
-    constants[1] = bij * 0.1;
+    *bij *= 0.1f;
```

Comment:
It's better to write this as `*bij /= 10`.  The reason is that 0.1 can't be exactly represented in floating point.  In double precision mode, you're multiplying by a single precision approximation to 0.1.  If you instead divide by an integer constant (exact in all modes), the compiler will automatically convert it to multiplying by an appropriate constant whose precision matches the precision of bij.